client/signup/steps/domains/index.jsx

```diff
@@ -690,23 +663,22 @@ class DomainsStep extends Component {
 		return this.props.isDomainOnly ? 'domain-first' : 'signup';
 	}
 
+	resetState() {
+		if ( inputMode.domainInput === this.state?.step ) {
```

Comment:
Perhaps you meant to access `this.props.step` or `this.state.currentStep` here? I don't see a `step` component state being set anywhere.